components/automate-cli/pkg/verifyserver/services/softwareversionservice/softwareversionservice_test.go

```diff
@@ -0,0 +1,438 @@
+						Passed:        false,
+						SuccessMsg:    "",
+						ErrorMsg:      "wrongcammand is not available",
+						ResolutionMsg: "Ensure wrongcammand is available in $PATH on the node",
+					},
+					{
+						Title:         "Linux Version Check",
+						Passed:        true,
+						SuccessMsg:    "Ubuntu version is 20.04",
+						ErrorMsg:      "",
+						ResolutionMsg: "",
+					},
+				},
+			},
+			expectedError: "",
+		},
+		{
+			description: "If the file is not avalable for OS Version ",
+			args: args{
+				query:      "postgres",
+				checkarray: checktrue,
+				osFilepath: failfilepath,
+			},
+			expectedBody: &models.SoftwareVersionDetails{
+				Passed: false,
+				Checks: []models.Checks{
+					{
+						Title:         "stat availability",
+						Passed:        true,
+						SuccessMsg:    "stat is available",
+						ErrorMsg:      "",
+						ResolutionMsg: "",
+					},
+					{
+						Title:         "mkdir availability",
+						Passed:        true,
+						SuccessMsg:    "mkdir is available",
+						ErrorMsg:      "",
+						ResolutionMsg: "",
+					},
+					{
+						Title:         "Linux Version Check",
+						Passed:        false,
+						SuccessMsg:    "",
+						ErrorMsg:      "Unable to get the os version from /etc/os-release file",
+						ResolutionMsg: "Ensure if the /etc/os-release is availabile on the path",
+					},
+				},
+			},
+			expectedError: "",
+		},
+		{
+			description: "If the os version is not supported by automate",
+			args: args{
+				query:      "postgres",
+				checkarray: checktrue,
+				osFilepath: failurefile,
+			},
+			expectedBody: &models.SoftwareVersionDetails{
+				Passed: false,
+				Checks: []models.Checks{
+					{
+						Title:         "stat availability",
+						Passed:        true,
+						SuccessMsg:    "stat is available",
+						ErrorMsg:      "",
+						ResolutionMsg: "",
+					},
+					{
+						Title:         "mkdir availability",
+						Passed:        true,
+						SuccessMsg:    "mkdir is available",
+						ErrorMsg:      "",
+						ResolutionMsg: "",
+					},
+					{
+						Title:         "Linux Version Check",
+						Passed:        false,
+						SuccessMsg:    "",
+						ErrorMsg:      "Debian GNU/Linux 10 (buster) version is not supported by automate",
+						ResolutionMsg: "Ensure Debian GNU/Linux 10 (buster) correct version is installed on the node",
+					},
+				},
+			},
+		},
+		{
+			description: "If the entered Query is not supported by us",
+			args: args{
+				query:      "wrongquery",
+				checkarray: checktrue,
+				osFilepath: successfile,
+			},
+			expectedBody: nil,
+			expectedError: "The query wrongquery is not supported. The Supported query's are = postgres, opensearch, bastion, automate, chef-server",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.description, func(t *testing.T) {
+			service.(*SoftwareVersionService).osFilepath = tt.args.osFilepath
+			service.(*SoftwareVersionService).cmdCheckArray = tt.args.checkarray
```

Comment:
Here we are expecting mkdir to be always present in the machine where the test is running, make PATH configurable so that we can test it anywhere according to need

Reply:
Done